### PR TITLE
[Depends on 62522] zephyr: cavs: add secondary core context save support

### PR DESCRIPTION
**Pull request by RanderWang**
Register pm_state_notifier to set ready_flag for secondary core when it is powered up for second time after first fw boot. We can remove CONFIG_ADSP_IMR_CONTEXT_SAVE check for cavs platform for this feature.

zephyr side pr: https://github.com/zephyrproject-rtos/zephyr/pull/62522

It has been discussed in https://github.com/thesofproject/sof/pull/8156 and https://github.com/thesofproject/sof/pull/8170

The IMR_CONTEXT_SAVE check was first introduced by https://github.com/thesofproject/sof/pull/8027

### CODE REVIEW

**Review by kv2019i — COMMENTED** [review]
Thanks @RanderWang ! Have you made a test PR that combines this PR with a Zephyr update pointing to your Zephyr patch (like I did https://github.com/thesofproject/sof/pull/8136 ). This would good step to make to ensure the combination passes all SOF CI tests.

The patch itself looks good. I won't mark approve yet to ensure this isn't accidentally merged too soon, but yes, looks good.

**Review by lgirdwood — APPROVED** [review]

**Comment by RanderWang** [comment]
> Thanks @RanderWang ! Have you made a test PR that combines this PR with a Zephyr update pointing to your Zephyr patch (like I did #8136 ). This would good step to make to ensure the combination passes all SOF CI tests.
> 
> The patch itself looks good. I won't mark approve yet to ensure this isn't accidentally merged too soon, but yes, looks good.

@kv2019i I tested it in local environment. Current Github CI test doesn't use multi-core topology so we can't depend on Github CI.
My next plan is to add multi-core support in cavs-nocodec and remove cavs-nocodec-multicore topology as we have discussed

File: zephyr/lib/cpu.c
Comment (lyakh) [inline zephyr/lib/cpu.c:151]: this means, that now if IMR context saving is **not** enabled we potentially call `z_init_cpu()` less frequently, or will it not happen because now the `pm_state_next_get(id)->state == PM_STATE_ACTIVE` will cover all those cases?

Comment (RanderWang) [reply zephyr/lib/cpu.c:151]: sure,  we only need to call z_init_cpu() for dsp core init with above condition for both CAVS  and ACE platforms since CAVS dsp core context save is being supported now.  IMR_CONTEXT_SAVE has nothing to do with z_init_cpu because
IMR_CONTEXT_SAVE is for the whole audio context but z_init_cpu is only for dsp core. IMR_CONTEXT_SAVE is not fit for this check but it can differentiate CAVS & ACE, so it was used.

**Review by tmleman — APPROVED** [review]

**Comment by lgirdwood** [comment]
@andyross need your re-review on the Zephyr dependecy to unblock this target for 3rd party SDV usage. Thanks !
@cujomalainey fyi.

**Comment by RanderWang** [comment]
@lgirdwood @kv2019i    I tested this PR with multi-core tplg https://github.com/thesofproject/sof/pull/8240 (include recent zephyr cavs multicore changes). All tests passed.  Next I will update the west for zephyr and try to merge the topology pr.  Thanks!

**Review by kv2019i — APPROVED** [review]
Looks good! I guess we can merge this already now, and merge the Zephyr part independently @RanderWang  ?

**Comment by RanderWang** [comment]
> Looks good! I guess we can merge this already now, and merge the Zephyr part independently @RanderWang ?

sure. Zephyr part has been merged. Thanks!

**Comment by kv2019i** [comment]
@RanderWang I now pushed https://github.com/thesofproject/sof/pull/8463 . The Zephyr change does not become visible to SOF until we bring it in via west.yml. Not sure whether we can merged this and #8463 independently, do we need to merge in particular order, or even at the same time...?

**Comment by RanderWang** [comment]
> @RanderWang I now pushed #8463 . The Zephyr change does not become visible to SOF until we bring it in via west.yml. Not sure whether we can merged this and #8463 independently, do we need to merge in particular order, or even at the same time...?

@kv2019i This PR is for multi-core feature and depends on topology support. Now our topology is not merged and waiting for this PR so CI doesn't get any error. (topology PR: https://github.com/thesofproject/sof/pull/8240)

**Pull request merged**
